### ISSUE TITLE
AddBar() should set fill color for negative bars

### ISSUE THREAD

**Issue by pietcoussens** [ScottPlot/ScottPlot#946]
Hello, 

In the attached project, I have 2 curves I want to plot in a Bar graph. When the value goes below 0, the bars colors red for both curves. I do not find how to get the same color for both positive and negative values.
I have zipped my Visual Studio C# project herewith.

Regards, Piet Coussens, Belgium.
[ScottPlot_testproject.zip](https://github.com/ScottPlot/ScottPlot/files/6259268/ScottPlot_testproject.zip)
In the picture the upper section uses the scatterplot-function while the lower one uses the AddBarGroups-function
![Naamloos](https://user-images.githubusercontent.com/81981517/113598999-471aa780-963e-11eb-9a05-b93ce08bd3ba.png)

**Comment by swharden** [comment]
Hi @pietcoussens,

This looks like a bug! There is a "feature" where negative bars can be given an alternate color, but it looks like it defaults to red but should probably default to the color of the positive bars such that the user must manually activate this feature. I'll fix this in the next release.

This topic is demonstrated by this cookbook recipe:
https://swharden.com/scottplot/cookbooks/4.1.11-beta/category/plottable-bar-graph/#negative-bar-colors

For now, you can probably fix your issue by doing something like this
```cs
var plt = new ScottPlot.Plot(600, 400);
double[] values = { 23, -17, 19, -24, 22 };
var bar = plt.AddBar(values);
bar.FillColorNegative = bar.FillColor; // make the negative color the same as the positive color
```

**Comment by pietcoussens** [comment]
Hello,

Thanks for the quick reply.
Unfortunately I have to use the AddBarGroups method because I have to plot 2 bars on the same drawing. When using the AddBar method both bars are drawn over each other and that does not do what I need.
The AddBarGroups method however does nog have the FillColor and FillColorNegative attribute.
I will have to wait for your next update I suppose.

Regards,              Piet Coussens

Van: Scott W Harden ***@***.***>
Verzonden: dinsdag 6 april 2021 3:32
Aan: ScottPlot/ScottPlot ***@***.***>
CC: Piet Coussens ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [ScottPlot/ScottPlot] When using AddBarGroups in 4.1.11 (beta), I only get red bars if the value is below 0 (#946)


Hi @pietcoussens<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpietcoussens&data=04%7C01%7Cpiet.coussens%40odisee.be%7Cf7b16dfe1b98421ad5eb08d8f89bb5b5%7C5e74901d334f46e396d147d842585abd%7C0%7C0%7C637532695389429160%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=D9pNvLm%2F22MEuEPrlWAVqNvdb3tp1vQlRfJQ9fODqBg%3D&reserved=0>,

This looks like a bug! There is a "feature" where negative bars can be given an alternate color, but it looks like it defaults to red but should probably default to the color of the positive bars such that the user must manually activate this feature. I'll fix this in the next release.

This topic is demonstrated by this cookbook recipe:
https://swharden.com/scottplot/cookbooks/4.1.11-beta/category/plottable-bar-graph/#negative-bar-colors<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fswharden.com%2Fscottplot%2Fcookbooks%2F4.1.11-beta%2Fcategory%2Fplottable-bar-graph%2F%23negative-bar-colors&data=04%7C01%7Cpiet.coussens%40odisee.be%7Cf7b16dfe1b98421ad5eb08d8f89bb5b5%7C5e74901d334f46e396d147d842585abd%7C0%7C0%7C637532695389429160%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=vPlZnusErMOlUaYpqea11Pggz6yxDW08bn2rZDB2uSQ%3D&reserved=0>

For now, you can probably fix your issue by doing something like this

var plt = new ScottPlot.Plot(600, 400);

double[] values = { 23, -17, 19, -24, 22 };

var bar = plt.AddBar(values);

bar.FillColorNegative = bar.FillColor; // make the negative color the same as the positive color

-

**Comment by swharden** [comment]
Fixed! This will be in the next release, hopefully this weekend 🤞

**Comment by pietcoussens** [comment]
Nice,

I am currently writing a set of lab exercises in C# for my students about FFT and your packages were a great help.

Regards, Piet Coussens


Van: Scott W Harden ***@***.***>
Verzonden: zondag 11 april 2021 0:56
Aan: ScottPlot/ScottPlot ***@***.***>
CC: Piet Coussens ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [ScottPlot/ScottPlot] AddBar() should set fill color for negative bars (#946)


Fixed! This will be in the next release, hopefully this weekend 🤞

**Comment by pietcoussens** [comment]
Nice,

I am currently writing a lab exercise in C# for my students about FFT and your packages were a great help.

Regards, Piet Coussens

Van: Scott W Harden ***@***.***>
Verzonden: zondag 11 april 2021 0:56
Aan: ScottPlot/ScottPlot ***@***.***>
CC: Piet Coussens ***@***.***>; Mention ***@***.***>
Onderwerp: Re: [ScottPlot/ScottPlot] AddBar() should set fill color for negative bars (#946)


Closed #946<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FScottPlot%2FScottPlot%2Fissues%2F946&data=04%7C01%7Cpiet.coussens%40odisee.be%7Cef610b08d2bf4b2ae26808d8fc73c251%7C5e74901d334f46e396d147d842585abd%7C0%7C0%7C637536921438602572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=mpcxiBn4KCzb2CPB41PLqnNrA3HhsqGT8nehW2qy1qE%3D&reserved=0> via #968<https://eur02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FScottPlot%2FScottPlot%2Fpull%2F968&data=04%7C01%7Cpiet.coussens%40odisee.be%7Cef610b08d2bf4b2ae26808d8fc73c251%7C5e74901d334f46e396d147d842585abd%7C0%7C0%7C637536921438602572%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=qhTGSmsupBDLCemVvRqVsgpOBEJjF22k863zNdXoH58%3D&reserved=0>.

-

**Comment by pietcoussens** [comment]
What I forgot in the previous mail:
The plot is defined in the file “RekenEnDisplay.cs” between the lines 145 and 154 (Plot2)

Regards,              Piet Coussens

Van: Piet Coussens
Verzonden: vrijdag 16 april 2021 10:09
Aan: ScottPlot/ScottPlot ***@***.***>
Onderwerp: RE: [ScottPlot/ScottPlot] AddBar() should set fill color for negative bars (#946)

Hello,

I regret to have to inform you that this problem does not seems to be solved in my case ( C# in Visual Studio) with release 4.1.12. Negative values are still red colored instead of the foreground color.
I have zipped my project and attached to this mail for reproducability.
Or am I doing something wrong?

Regards, Piet Coussens

Van: Piet Coussens
Verzonden: zondag 11 april 2021 16:05
Aan: ScottPlot/ScottPlot ***@***.******@***.***>>
Onderwerp: RE: [ScottPlot/ScottPlot] AddBar() should set fill color for negative bars (#946)

Nice,

I am currently writing a set of lab exercises in C# for my students about FFT and your packages were a great help.

Regards, Piet Coussens


Van: Scott W Harden ***@***.******@***.***>>
Verzonden: zondag 11 april 2021 0:56
Aan: ScottPlot/ScottPlot ***@***.******@***.***>>
CC: Piet Coussens ***@***.******@***.***>>; Mention ***@***.******@***.***>>
Onderwerp: Re: [ScottPlot/ScottPlot] AddBar() should set fill color for negative bars (#946)


Fixed! This will be in the next release, hopefully this weekend 🤞
